java/flight/src/main/java/org/apache/arrow/flight/FlightClient.java

```diff
@@ -127,7 +127,7 @@ public ClientStreamListener startPut(FlightDescriptor descriptor, VectorSchemaRo
     // send the schema to start.
     ArrowMessage message = new ArrowMessage(descriptor.toProtocol(), root.getSchema());
     observer.onNext(message);
-    return new PutObserver(new VectorUnloader(root, true, false), observer, resultObserver.getFuture());
+    return new PutObserver(new VectorUnloader(root, true, true), observer, resultObserver.getFuture());
```

Comment:
I'm not competent to comment on Java code, but argument lists such as `true, false` or `true, true` strike me as cryptic. Is it possible to mention the argument name somehow? For example `new VectorUnloader(root, true /* foo */, true /* bar */)`.

Reply:
I'd say it's not good practice to have so many flags (and a builder could be used instead), though some IDEs get around this by displaying the argument name inline for literal value arguments. I'll add comments since the second `true` is quite important though.